squarelet/users/models.py

```diff
@@ -218,6 +219,32 @@ def is_hub_eligible(self):
             | Q(parent__hub_eligible=True)
         ).exists()
 
+    def get_potential_organizations(self):
+        # Get all verified email addresses for the user
+        emails = self.emailaddress_set.filter(verified=True)
+
+        # Extract the domains from the verified emails
+        domains = [email.email.split("@")[1].lower() for email in emails]
+
+        # Find organizations matching any of the domains
+        return (
+            Organization.objects.filter(domains__domain__in=domains)
+            .distinct()
+            .exclude(users=self)
+        )
+
+    def can_auto_join(self, organization):
+        """Check if the user can auto-join an organization."""
+        # Get the verified email addresses for the user
+        user_emails = self.emailaddress_set.filter(verified=True)
+        user_domains = [email.email.split("@")[1].lower() for email in user_emails]
+
+        # Get the organization's verified domains
+        org_domains = [domain.domain.lower() for domain in organization.domains.all()]
+
+        # Check if any of the user's domains match the organization's domains
+        return any(domain in org_domains for domain in user_domains)
```

Comment:
Instead of duplicating the functionality above to get the user's verified emails and extract domains from them, we can test whether the `organization` argument is in the result from `get_potential_organizations()`. Alternatively, could we add a function like `get_user_domains` that can be tested and reused by both `get_potential_organizations` and `can_auto_join`.